lib/temporal/client.rb

```diff
@@ -229,25 +229,29 @@ def await_workflow_result(workflow, workflow_id:, run_id: nil, timeout: nil, nam
       execution_options = ExecutionOptions.new(workflow, options, config.default_execution_options)
       max_timeout = Temporal::Connection::GRPC::SERVER_MAX_GET_WORKFLOW_EXECUTION_HISTORY_POLL
       history_response = nil
-      begin
-        history_response = connection.get_workflow_execution_history(
-          namespace: execution_options.namespace,
-          workflow_id: workflow_id,
-          run_id: run_id,
-          wait_for_new_event: true,
-          event_type: :close,
-          timeout: timeout || max_timeout,
-        )
-      rescue GRPC::DeadlineExceeded => e
-        message = if timeout
-          "Timed out after your specified limit of timeout: #{timeout} seconds"
-        else
-          "Timed out after #{max_timeout} seconds, which is the maximum supported amount."
+      while(true) do
+        begin
+          history_response = connection.get_workflow_execution_history(
+            namespace: execution_options.namespace,
+            workflow_id: workflow_id,
+            run_id: run_id,
+            wait_for_new_event: true,
+            event_type: :close,
+            timeout: timeout || max_timeout,
+          )
+        rescue GRPC::DeadlineExceeded => e
+          message = if timeout
+            "Timed out after your specified limit of timeout: #{timeout} seconds"
+          else
+            "Timed out after #{max_timeout} seconds, which is the maximum supported amount."
+          end
+          raise TimeoutError.new(message)
```

Comment:
If the plan is to bring this in line with the other SDKs, should this be a `retry` rather than a `raise`? Maybe with support for a max retry count that defaults to `1` for backwards compatibility?

Reply:
TypeScript SDK does not have retries for timeout. I think it should be left to the client to decide on how they want to deal with it.

Reply:
I think it should follow the behavior of the other SDKs, but that could trip up any current code that assumes that calling `await_workflow_result` will succeed or time out.

Put another way, if code no longer times out, current users may end up with some nasty latency surprises. And since this gem's version doesn't get bumped (or officially published; #285) very often, I don't know if there'd be a good way to signal a big breaking change like that 😬

Reply:
Whoops, I get what you're saying now. I overlooked that the TS SDK also raises on deadline exceeded errors, it just hides it behind the `rethrowGrpcError` method 👍